### DISCUSSION TITLE
doctrine and no-warmup option

### DISCUSSION THREAD

**Discussion by pcmanprogrammeur** [symfony/symfony#48150] [Q&A] [answered]
### Symfony version(s) affected

5.4

### Description

Hi all,

I have a problem with the option "--no-warmup" in my composer.json to avoid doctrine warmup.

### How to reproduce

In my project, i have doctrine and when i want to build my docker image in production environment, the DNS is not defined so "my_server.my_domain" is unknown.

The problem is that when i launch a composer install, it automatically clear the cache and doctrine want to connect to my database.

To avoid it, i tried to add `--no-warmup` in my composer.json
`"cache:clear --no-warmup": "symfony-cmd",`

But as you can see, one time it works, one time it doesn't work, one time it works, one time it doesn't work, ....

```
/var/www/app # php bin/console cache:clear  --no-warmup

 // Clearing the cache for the prod environment with debug false                                                        
 [OK] Cache for the "prod" environment (debug=false) was successfully cleared.                                          

/var/www/app # php bin/console cache:clear  --no-warmup

In ExceptionConverter.php line 82:
  An exception occurred in the driver: SQLSTATE[08006] [7] could not translate host name "my_server.my_domain" to address: Name does not resolve  
                                                                                                                                   
In Exception.php line 30:
  SQLSTATE[08006] [7] could not translate host name "my_server.my_domain" to address: Name does not resolve  

In Driver.php line 28:
  SQLSTATE[08006] [7] could not translate host name "my_server.my_domain" to address: Name does not resolve  

/var/www/app # php bin/console cache:clear  --no-warmup

 // Clearing the cache for the prod environment with debug false                                                        
 [OK] Cache for the "prod" environment (debug=false) was successfully cleared.                                          

/var/www/app #
```

So, i have to two questions :
- Why the no-warmup option doesn't work each time, please ?
- Is there an other solution to avoid the default warmup of doctrine, please ?

Thanks for your help !

**Comment by derrabus** [comment, marked as answer]
When generating entity proxies, the ORM tries to establish a database connection to determine which database version is being used. To avoid this, set the server version explicitly in your Doctrine bundle configuration, e.g. `server_version:   '8.0.33' if your server is a MySQL 8.0.33.

**Reply by pcmanprogrammeur** [reply]
Thank you very much ! It works perfecty !
